src/app/components/search/search.component.ts

```diff
@@ -0,0 +1,38 @@
+import { Component, OnInit } from '@angular/core';
+import { debounceTime, filter, tap } from 'rxjs';
+import { SearchApiService } from './search-api.service';
+import { SearchResponse } from './types/search-response';
+import { FormControl } from '@angular/forms';
+
+@Component({
+  selector: 'mdb-search',
+  templateUrl: './search.component.html',
+  styleUrls: ['./search.component.scss'],
+})
+export class SearchComponent implements OnInit {
+  constructor(private searchApi: SearchApiService) {}
+
+  public searchInput = new FormControl();
+  public searchResult?: SearchResponse;
```

Comment:
I think you can do it in more elegant way. Let's create two Observable variables:
- check counts of symbols which user input
- second search results

And after just use `async` pipe to subscribe for the data inside the template

Something like that

```
  showSearch$: Observable<boolean> = this.searchInput.valueChanges.pipe(
    map((value) => (value ?? '').length > 3)
  );


  searchResults$: Observable<MediaItem[]> = this.searchInput.valueChanges.pipe(
    filter(Boolean),
    filter((value) => value.length > 3),
    debounceTime(200),
    distinctUntilChanged(),
    switchMap((value) => this.searchApi.requestSearchResult(value))
  );
```

Reply:
Reworked logic as per suggested changes.